Dev add company resource

emuzykin opened the pull request:
This is to pull maximum concurrent svms info

cchristous commented:
I believe it is a requirement for this repo. I noticed inconsistent spacing in the ```Company``` members

ferrants commented:
Not required, but i think it's worth doing since this is an open library. We should probably actually have a jenkins job that runs pylint and the tests. @careyrp - thoughts?

emuzykin commented:
I fixed the space issue. Any other comments or is this good to be shipped?

careyrp commented:
As we talked about yesterday the properties on the Company class do not work. The data structure is more complex.

ferrants commented:
Company class should have a test, right?

@careyrp - what do you think about having a job that can run the tests on this repo?

emuzykin commented:
I added a dictionary of dictionaries to pull company resource information. This was Ryan's vision for this change. @ferrants - i added a test for the Company class as well.
Test passed:
test_company_quota (test.data_test.TestData) ... ok

cchristous commented on dxskytap/company.py:
```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2013, DataXu Inc.
```
The copyright year should be 2014

cchristous commented on dxskytap/company.py:
```diff
@@ -0,0 +1,69 @@
+class Company_Quota(RestObject):
+    def __init__(self, connect, res, initial_data, parent):
+        super(Company_Quota, self).__init__(connect, res, initial_data,
+            is_full=True) # parent=parent, parent_attr="hardware"
```
please remove commented out code

emuzykin commented:
@cchristous - updated per Pylint feedback

cchristous commented:
thanks

ferrants commented:
Tests passed for me. :shipit: . BTW - This is what i got from pylint:
```
$ pylint dxskytap/company.py
No config file found, using default configuration
************* Module dxskytap.company
C:  3, 0: Trailing whitespace (trailing-whitespace)
C: 14, 0: Trailing whitespace (trailing-whitespace)
C: 15, 0: Trailing whitespace (trailing-whitespace)
C: 40, 0: Line too long (109/100) (line-too-long)
C: 41, 0: Wrong continued indentation.
        'cumulative_svms', 'concurrent_public_ips', 'concurrent_networks')
        ^                                     | (bad-continuation)
C: 49, 0: Trailing whitespace (trailing-whitespace)
C:  1, 0: Missing module docstring (missing-docstring)
C: 28, 0: Missing class docstring (missing-docstring)
C: 32, 4: Missing method docstring (missing-docstring)
R: 28, 0: Too few public methods (1/2) (too-few-public-methods)
C: 36, 0: Invalid class name "Company_Quota" (invalid-name)
C: 36, 0: Missing class docstring (missing-docstring)
C: 51, 4: Missing method docstring (missing-docstring)
R: 36, 0: Too few public methods (1/2) (too-few-public-methods)


Report
======
30 statements analysed.

Statistics by type
------------------

+---------+-------+-----------+-----------+------------+---------+
|type     |number |old number |difference |%documented |%badname |
+=========+=======+===========+===========+============+=========+
|module   |1      |1          |=          |0.00        |0.00     |
+---------+-------+-----------+-----------+------------+---------+
|class    |2      |2          |=          |0.00        |50.00    |
+---------+-------+-----------+-----------+------------+---------+
|method   |4      |4          |=          |50.00       |0.00     |
+---------+-------+-----------+-----------+------------+---------+
|function |0      |0          |=          |0           |0        |
+---------+-------+-----------+-----------+------------+---------+



External dependencies
---------------------
::

    dxskytap 
      \-restobject (dxskytap.company)



Raw metrics
-----------

+----------+-------+------+---------+-----------+
|type      |number |%     |previous |difference |
+==========+=======+======+=========+===========+
|code      |30     |52.63 |32       |-2.00      |
+----------+-------+------+---------+-----------+
|docstring |1      |1.75  |2        |-1.00      |
+----------+-------+------+---------+-----------+
|comment   |24     |42.11 |24       |=          |
+----------+-------+------+---------+-----------+
|empty     |2      |3.51  |2        |=          |
+----------+-------+------+---------+-----------+



Duplication
-----------

+-------------------------+------+---------+-----------+
|                         |now   |previous |difference |
+=========================+======+=========+===========+
|nb duplicated lines      |0     |0        |=          |
+-------------------------+------+---------+-----------+
|percent duplicated lines |0.000 |0.000    |=          |
+-------------------------+------+---------+-----------+



Messages by category
--------------------

+-----------+-------+---------+-----------+
|type       |number |previous |difference |
+===========+=======+=========+===========+
|convention |12     |21       |-9.00      |
+-----------+-------+---------+-----------+
|refactor   |2      |2        |=          |
+-----------+-------+---------+-----------+
|warning    |0      |2        |-2.00      |
+-----------+-------+---------+-----------+
|error      |0      |0        |=          |
+-----------+-------+---------+-----------+



Messages
--------

+-----------------------+------------+
|message id             |occurrences |
+=======================+============+
|missing-docstring      |5           |
+-----------------------+------------+
|trailing-whitespace    |4           |
+-----------------------+------------+
|too-few-public-methods |2           |
+-----------------------+------------+
|line-too-long          |1           |
+-----------------------+------------+
|invalid-name           |1           |
+-----------------------+------------+
|bad-continuation       |1           |
+-----------------------+------------+



Global evaluation
-----------------
Your code has been rated at 5.33/10 (previous run: 1.67/10, +3.67)

```
